entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -502,6 +502,46 @@ class Neo4jRepository(
         )
     }
 
+    fun getEntityTypes(): List<Map<String, Any>> {
+        // The match on geoProperties is specific since they are not fully supported
+        // (currently we only support location that is stored among the entity node attributes)
+        val query =
+            """
+                MATCH (entity:Entity)
+                OPTIONAL MATCH (entity)-[:HAS_VALUE]->(property:Property)
+                OPTIONAL MATCH (entity)-[:HAS_OBJECT]->(rel:Relationship)
+                RETURN labels(entity) as entityType, collect(distinct property.name) as propertyNames, 
+                    reduce(output = [], r IN collect(distinct labels(rel)) | output + r) as relationshipNames,
+                    count(entity.location) as entityWithLocationCount
+            """.trimIndent()
+
+        val result = session.query(query, emptyMap<String, Any>(), true).toList()
+        return result.map {
```

Comment:
what about filtering `User`, `Client` and `Group` entiities?

Reply:
Indeed, should we also filter the `specificAccessPolicy` property ?

Reply:
hmm, no ...